src/studylist/StudyListWithData.js

```diff
@@ -129,28 +162,72 @@ class StudyListWithData extends Component {
     this.searchForStudies(searchData);
   };
 
+  update = () => {
+    this.setState({
+      modalComponentId: null,
+      showStudyList: true,
+    });
+  };
+
   render() {
     if (this.state.error) {
       return <div>Error: {JSON.stringify(this.state.error)}</div>;
-    } else if (this.state.studies === null) {
+    } else if (this.state.studies === null && !this.state.modalComponentId) {
       return <div>Loading...</div>;
     }
 
+    let healthCare = '';
+    if (this.props.user) {
+      let modalContent = '';
+      if (this.state.modalComponentId === 'DicomStorePicker') {
+        modalContent = <ConnectedDicomStorePicker onClose={this.update} />;
+      } else if (this.state.modalComponentId === 'DicomFilesUploader') {
+        modalContent = <ConnectedDicomFilesUploader onClose={this.update} />;
+      }
+      healthCare = (
```

Comment:
After this PR is merged: https://github.com/OHIF/react-viewerbase/pull/129 we can put these buttons in the children of the <StudyList> component. That will make this a bit cleaner, since right now they are just floating.

Reply:
done